test/local_sensitivity/callbacks.jl

```diff
@@ -1,5 +1,178 @@
-using StochasticDiffEq, Zygote
-using DiffEqSensitivity, Test
+using StochasticDiffEq, OrdinaryDiffEq, Zygote
+using DiffEqSensitivity, Test, ForwardDiff
+
+# ODEs
+
+function fiip(du,u,p,t)
+  du[1] = dx = p[1]*u[1] - p[2]*u[1]*u[2]
+  du[2] = dy = -p[3]*u[2] + p[4]*u[1]*u[2]
+end
+function foop(u,p,t)
+  dx = p[1]*u[1] - p[2]*u[1]*u[2]
+  dy = -p[3]*u[2] + p[4]*u[1]*u[2]
+  [dx,dy]
+end
+
+p = [1.5,1.0,3.0,1.0]; u0 = [1.0;1.0]
+
+prob = ODEProblem(fiip,u0,(0.0,10.0),p)
+proboop = ODEProblem(foop,u0,(0.0,10.0),p)
+
+# callback with no effect
+
+condition(u,t,integrator) = t == 5
+affect!(integrator) = integrator.u[1] += 0.0
+cb = DiscreteCallback(condition,affect!,save_positions=(false,false))
+
+sol1 = solve(prob,Tsit5(),u0=u0,p=p,callback=cb,tstops=[5.0],abstol=1e-14,reltol=1e-14,saveat=0.1)
+sol2 = solve(prob,Tsit5(),u0=u0,p=p,tstops=[5.0],abstol=1e-14,reltol=1e-14,saveat=0.1)
+
+@test sol1.u == sol2.u
+
+du01,dp1 = Zygote.gradient(
+  (u0,p)->sum(solve(prob,Tsit5(),u0=u0,p=p,callback=cb,tstops=[5.0],abstol=1e-14,reltol=1e-14,saveat=0.1,sensealg=BacksolveAdjoint())),
+  u0,p)
+
+du01a,dp1a = Zygote.gradient(
+  (u0,p)->sum(solve(prob,Tsit5(),u0=u0,p=p,tstops=[5.0],abstol=1e-14,reltol=1e-14,saveat=0.1,sensealg=BacksolveAdjoint())),
+  u0,p)
+
+du02,dp2 = Zygote.gradient(
+  (u0,p)->sum(solve(prob,Tsit5(),u0=u0,p=p,callback=cb,tstops=[5.0],abstol=1e-14,reltol=1e-14,saveat=0.1,sensealg=ReverseDiffAdjoint()))
+  ,u0,p)
+
+dstuff = ForwardDiff.gradient(
+  (θ)->sum(solve(prob,Tsit5(),u0=θ[1:2],p=θ[3:6],callback=cb,tstops=[5.0],abstol=1e-14,reltol=1e-14,saveat=0.1)),
+  [u0;p])
+
+@test du01 ≈ dstuff[1:2]
+@test dp1 ≈ dstuff[3:6]
+@test du01a ≈ dstuff[1:2]
+@test dp1a ≈ dstuff[3:6]
+@test du01 ≈ du02
+@test dp1 ≈ dp2
+
+@test du02 ≈ dstuff[1:2]
+@test dp2 ≈ dstuff[3:6]
+
+
+# callback with no effect except saving the state
+
+condition(u,t,integrator) = t == 5
+affect!(integrator) = integrator.u[1] += 0.0
+cb = DiscreteCallback(condition,affect!)
```

Comment:
just found that this test is already broken at the moment, i.e., `save_positions=(true,true)` instead of `save_positions=(false,false)`.

Reply:
Should we internally change save_positions or something? That might give odd behavior.